common/src/address/mod.rs

```diff
@@ -129,6 +131,12 @@ impl<T: Addressable> Address<T> {
     }
 }
 
+impl<T> Address<T> {
+    pub fn into_string(self) -> String {
+        self.address
+    }
```

Comment:
This method is a problem. The problem is that we do not assume that the address in `Address<T>` is a valid address. Hence, you could get a completely bogus address out of this. Actually, the fact that the address can be invalid is part of why RpcAddress works, because the checking is done afterwards, with decode_object.

Now this isn't a problem I know how to solve. Perhaps we should remove the Display impl and only keep your into_string, and make it return a `Result<String>`... but I really don't know the best way to do this.